crates/html_node/src/lib.rs

```diff
@@ -94,11 +108,22 @@ fn minify_inner(code: &str, opts: MinifyOptions) -> anyhow::Result<String> {
             }
         };
 
+        let mut returned_errors = None;
+
         if !errors.is_empty() {
+            let mut internal_errors = Vec::with_capacity(errors.len());
+
             for err in errors {
-                err.to_diagnostics(handler).emit();
+                let mut buf = vec![];
+
+                err.to_diagnostics(handler).buffer(&mut buf);
+
+                for i in buf {
+                    internal_errors.push(Error { text: i.message() });
```

Comment:
I found it very useful, so I ahve question here:
- What is logic in `swc`? How it returns errors and where I can get structure?
- Why don't use `swc_common::errors::Diagnostic` and add `Serialize`? So developer can get any possible infromation?

I think the second variant is really good

Because in most of cases developer need not only `text`, also - location, suggestions, notes and etc

Reply:
- You can get errors using `Handler`
 - I think `swc_common::errors::Diagnostic` is better but I simply copy pasted some code, so I don't know the reasoning behind this code.